webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+                if (responseForUser == null || responseForUser.isBlank()) {
+                    return Maybe.just(
+                        Content.builder()
+                            .role("model")
+                            .parts(List.of(Part.fromText(UNABLE_TO_GENERATE_FLOW_ERROR)))
+                            .build()
+                    );
+                }
+
+                return Maybe.just(
+                    Content.builder()
+                        .role("model")
+                        .parts(List.of(Part.fromText(responseForUser)))
+                        .build()
+                );
+            })
+            .build();
+    }
+
+    // Utility to minify a JSON schema by removing unnecessary fields
+    private static void minifySchema(JsonNode node) {
+        if (node == null) return;
+        if (node.isObject()) {
+            ObjectNode obj = (ObjectNode) node;
+            obj.remove("$dynamic");
+            obj.remove("$group");
+            if (obj.optional("default").map(d -> d.isBoolean() && !d.asBoolean()).orElse(false)) {
+                obj.remove("default");
+            }
+            obj.properties().forEach(e -> minifySchema(e.getValue()));
+        } else if (node.isArray()) {
+            ArrayNode arr = (ArrayNode) node;
+            for (JsonNode item : arr) {
+                minifySchema(item);
+            }
+        }
+    }
+
+    private static Maybe<Content> shortCircuitLlmResponse(CallbackContext callbackContext, String response) {
+        callbackContext.eventActions().setEscalate(true);
+        return Maybe.just(Content.builder()
+            .role("model")
+            .parts(List.of(Part.fromText(response)))
+            .build()
+        );
+    }
+
+    @Override
+    public Mono<String> generateFlow(FlowGenerationPrompt flowGenerationPrompt) {
+        String generatedUserId = FriendlyId.createFriendlyId();
+        Content userMsg = Content.fromParts(
+            Part.fromText("Current Flow YAML:\n```yaml\n" + flowGenerationPrompt.flowYaml() + "\n```"),
+            Part.fromText("Build a Flow that does the following:\n```\n" + flowGenerationPrompt.userPrompt() + "\n```")
+        );
+
+        AtomicReference<Session> sessionRef = new AtomicReference<>();
+        return Mono.from(
+            IN_MEMORY_RUNNER.sessionService().createSession(KESTRA_FLOW_AGENT_APP_ID, generatedUserId)
+                .doOnSuccess(sessionRef::set)
+                .flatMapPublisher(session -> IN_MEMORY_RUNNER.runAsync(session, userMsg, RunConfig.builder().build()))
+                .filter(e -> e.finalResponse() && e.author().equals(KESTRA_FLOW_BUILDER_NAME))
+                .map(Event::stringifyContent)
+                .reduce("", (acc, content) -> acc + content)
+                .doOnSuccess(s -> {
+                    if (POSSIBLE_ERROR_MESSAGES.contains(s)) {
+                        throw new AiException(s);
+                    }
```

Comment:
Error seems to be part of the normal process (not an exceptional path) so I'd rather use a responseType of something else. But I'm also OK with that if you prefer.